### PR TITLE
mingw-w64: add missing CRT sources

### PR DESCRIPTION
**Pull request by ypsvlq**
fixes https://github.com/ziglang/zig/issues/9364, fixes https://github.com/ziglang/zig/issues/18181

### CODE REVIEW

**Review by andrewrk — CHANGES_REQUESTED** [review]
C files in the include directory?

**Comment by ypsvlq** [comment]
@andrewrk this is the correct location, it can be verified against an existing mingw install and is briefly mentioned in MS docs [here](https://learn.microsoft.com/en-us/windows/win32/api/tsuserex/nn-tsuserex-iadstsuserex) (although that was the only file that seemed to be documented or even mentioned online, so leaving them out probably wouldn't cause any problems)

File: src/mingw.zig [hunk 1]
```diff
@@ -15,7 +15,66 @@ pub const CRTFile = enum {
...
+    ksguid,
+    locationapi,
+    sapi,
+    sensorsapi,
+    portabledeviceguids,
+    taskschd,
+    dloadhelper,
+    strmiids,
+    gdiplus,
+    adsiid,
```
Comment (andrewrk) [inline src/mingw.zig:45]: What's the purpose of adding all these additional CRTFile entries? Can't we just toss the source files into mingwex.lib?

**Review by andrewrk — APPROVED** [review]
Nice! Thanks for working with me on this. I'm happy with this. I hope you are too.

**Comment by ypsvlq** [comment]
Definitely, there's still some things I might look at in future but this gets it to a much better place both for usability and deciding what to add/remove.

**Comment by andrewrk** [comment]
I verified that running `zig build update-mingw` created an empty diff (copied files in this branch match upstream mingw-w64 exactly).

**Pull request merged**